web/app/js/components/util/PageVisibility.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import _isFunction from 'lodash/isFunction';
+import { usePageVisibility } from 'react-page-visibility';
+
+export function handlePageVisibility(params) {
+  const { prevVisibilityState, currentVisibilityState, onVisible, onHidden } = params;
+  if (prevVisibilityState !== undefined && currentVisibilityState !== undefined) {
```

Comment:
Thanks @rmars for your feedback. I think it’s a good idea to return from the function early if parameters are not defined. But thinking about the functions, perhaps it would be good to validate if they exist or not in the condition due to perhaps, in some cases, we need to use only one of the methods and not both. Imagine, for example, that we want to display a modal when page is not visible and keep there until the page is visible again to force user to close it manually. Please, let me know your thoughts.

Reply:
no problem!
yeah I can see that it might be confusing if `handlePageVisibility` returns if you don't supply both functions, in the case that the user supplied one of `onVisible` or `on Hidden` and expects to see that function called. I'm fine with putting the `isFunction` checks back to their original spots!

if we do end up wanting to force all users to always supply both a `onVisible` and `onHidden` then we can move those checks back to the top if.